pkg/envoy/sds/response.go

```diff
@@ -145,58 +144,66 @@ func (s *sdsImpl) getRootCert(cert certificate.Certificater, sdscert envoy.SDSCe
 		return secret, nil
 	}
 
-	// Program SAN matching based on SMI TrafficTarget policies
-	switch sdscert.CertType {
-	case envoy.RootCertTypeForMTLSOutbound:
+	svcIdentities, err := getServiceIdentitiesFromCert(sdsCert, s.serviceIdentity, s.meshCatalog)
+	if err != nil {
+		return nil, err
+	}
+
+	secret.GetValidationContext().MatchSubjectAltNames = getSubjectAltNamesFromSvcIdentities(svcIdentities)
+	return secret, nil
+}
+
+func getServiceIdentitiesFromCert(sdsCert envoy.SDSCert, svcIdentity identity.ServiceIdentity, mc catalog.MeshCataloger) ([]identity.ServiceIdentity, error) {
```

Comment:
Does this function return the identity as an FQDN or the ServiceAccount encapsulated as a ServiceIdentity type?
A comment will help here with an example of what the return value looks like, otherwise, the code is pretty hard to follow.

Reply:
Agreed the area of the code is hard to follow. That's why this needs refactoring.  This is step 1.  We'll continue to plow through this.  I have a few other ideas to offer.

Added a comment for the short time while this function is in existence.